packages/app/server/api/repo/search.get.ts

```diff
@@ -18,55 +50,86 @@ export default defineEventHandler(async (event) => {
-        prefix: usePackagesBucket.base,
-        limit: 1000,
-        cursor,
-      });
-      const { objects, truncated } = listResult;
-      cursor = truncated ? listResult.cursor : undefined;
-
-      for (const obj of objects) {
-        const parts = parseKey(obj.key);
-        const orgRepo = `${parts.org}/${parts.repo}`.toLowerCase();
-        const applies =
-          parts.org.toLowerCase().includes(searchText) ||
-          parts.repo.toLowerCase().includes(searchText) ||
-          orgRepo.includes(searchText);
-        if (!applies) continue;
-
-        const key = `${parts.org}/${parts.repo}`;
-        if (!seen.has(key)) {
-          seen.add(key);
-          uniqueNodes.push({
-            name: parts.repo,
-            owner: {
-              login: parts.org,
-              avatarUrl: `https://github.com/${parts.org}.png`,
-            },
-          });
-          if (uniqueNodes.length >= maxNodes) break;
+    // console.log(`[SEARCH-STREAM] Starting search for "${query.text}"`);
+
+    return streamResponse(event, async (stream) => {
+      const searchText = query.text.toLowerCase();
+      let cursor: string | undefined;
+      const seen = new Set<string>();
+      const maxNodes = 10;
+      let sentCount = 0;
+      let keepGoing = true;
+      let batchCount = 0;
+      let totalScanned = 0;
+
+      // console.log(
+      //   `[SEARCH-STREAM] Streaming response initiated for "${query.text}"`,
+      // );
+
+      while (sentCount < maxNodes && keepGoing && !signal.aborted) {
+        batchCount++;
+        const listResult = await r2Binding.list({
+          prefix: usePackagesBucket.base,
+          limit: 1000,
+          cursor,
+        });
+        const { objects, truncated } = listResult;
+        totalScanned += objects.length;
+        cursor = truncated ? listResult.cursor : undefined;
+
+        // console.log(
+        //   `[SEARCH-STREAM] Batch ${batchCount}: Scanned ${objects.length} objects, total ${totalScanned}`,
+        // );
+
+        for (const obj of objects) {
+          if (signal.aborted) break;
+
+          const parts = parseKey(obj.key);
+          const orgRepo = `${parts.org}/${parts.repo}`.toLowerCase();
+          const applies =
+            parts.org.toLowerCase().includes(searchText) ||
+            parts.repo.toLowerCase().includes(searchText) ||
+            orgRepo.includes(searchText);
+
+          if (!applies) continue;
+
+          const key = `${parts.org}/${parts.repo}`;
+          if (!seen.has(key)) {
+            seen.add(key);
+            const node = {
+              id: key,
+              name: parts.repo,
+              owner: {
+                login: parts.org,
+                avatarUrl: `https://github.com/${parts.org}.png`,
+              },
+              // search stats with each result
+              _stats: {
+                batchCount,
+                scannedSoFar: totalScanned,
+                matchNumber: sentCount + 1,
```

Comment:
why do we need matchNumber? and scannedSoFar? i do not think we need these, and also the batchCount.